server/core/terraform/terraform_client.go

```diff
@@ -283,30 +284,37 @@ func (c *DefaultClient) TerraformBinDir() string {
 func (c *DefaultClient) ListAvailableVersions(log logging.SimpleLogging) ([]string, error) {
 	url := fmt.Sprintf("%s/terraform", c.downloadBaseURL)
 
-	if !c.downloadsAllowed {
+	if !c.downloadAllowed {
 		log.Debug("Terraform downloads disabled. Won't list Terraform versions available at %s", url)
 		return []string{}, nil
 	}
 
 	log.Debug("Listing Terraform versions available at: %s", url)
+	// terraform-switcher calls os.Exit(1) if it fails to successfully GET the configured URL.
+	// So, before calling it, test if we can connect. Then we can return an error instead if the request fails.
+	resp, err := http.Get(url)
+	if err != nil || resp.StatusCode != 200 {
+		return nil, fmt.Errorf("Unable to list Terraform versions: %s", err)
+	}
```

Comment:
I've added a `#nosec` comment here for this particular rule (`G107`) because terraform-switcher makes (AFAICT) [the exact same call](https://github.com/warrensbox/terraform-switcher/blob/3072cc27e6725ea2e428202334b18d0144a9901a/lib/list_versions.go#L115) a few lines later. I guess that since their call is in a library and this one is not, it makes gosec unhappy.

I'm open to alternative suggestions if you have any.

Reply:
unsure what the best strategy is here.

Is there an alternative beside setting the nosec comment?

cc @runatlantis/maintainers

Reply:
G107 handle untrusted url, but this url is provided by server configuration. so it's ok to ignore it.